src/pages/loading-waiting-states/progress-bar/index.tsx

```diff
@@ -0,0 +1,234 @@
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import MenuIcon from '@material-ui/icons/Menu';
+import { Folder, VerticalAlignTop, Description } from '@material-ui/icons';
+import LinearProgress, { LinearProgressProps } from '@material-ui/core/LinearProgress';
+import { useDispatch } from 'react-redux';
+import { TOGGLE_DRAWER } from '../../../redux/actions';
+import { InfoListItem } from '@pxblue/react-components';
+import * as Colors from '@pxblue/colors';
+type FolderItem = {
+    id: string;
+    name: string;
+    progress: number;
+    status: string;
+};
+const foldersList = [
+    { label: 'The Best Dev Team', value: '1' },
+    { label: 'The Best Design Team', value: '2' },
+    { label: 'The Best UX Team', value: '3' },
+    { label: 'The Best Management Team', value: '4' },
+    { label: 'The Best Facility Team', value: '5' },
+    { label: 'The The Proudest Team', value: '6' },
+];
+const LinearProgressWithLabel = (props: LinearProgressProps & { value: number }): JSX.Element => (
+    <LinearProgress variant="determinate" {...props} />
+);
+const uploadFileList: FolderItem[] = [];
+
+const useStyles = makeStyles((theme: Theme) => ({
+    container: {
+        padding: `${theme.spacing(2)}px ${theme.spacing(2)}px`,
+        maxWidth: 600,
+        margin: '0 auto',
+    },
+    uploadButtonContainer: {
+        textAlign: 'right',
+        paddingBottom: theme.spacing(2),
+    },
+    closeButtonContainer: {
+        color: Colors.black[50],
+        borderColor: Colors.black[50],
+    },
+    cardContentStyle: {
+        padding: 0,
+    },
+    formControlStyle: {
+        width: '100%',
+    },
+    radioLabel: {
+        display: 'flex',
+    },
+    iconStyle: {
+        fill: Colors.gray[500],
+        margin: `0 ${theme.spacing(2)}px 0 ${theme.spacing(1)}px`,
+    },
+    formLabel: {
+        margin: 0,
+        width: '100%',
+        padding: theme.spacing(1),
+        borderBottom: `1px solid ${Colors.gray[50]}`,
+        '&:last-child': {
+            borderBottom: 'none',
+        },
+    },
+    toolbarGutters: {
+        padding: `0 ${theme.spacing(2)}px`,
+    },
+    placementOfList: {
+        position: 'absolute',
+        bottom: theme.spacing(3),
+        right: theme.spacing(3),
+    },
+    infoListItemStyle: {
+        marginBottom: theme.spacing(2),
+        '&:last-child': {
+            marginBottom: 0,
+        },
+    },
+    subTitle: {
+        color: Colors.black[200],
+    },
+}));
+const createFileItem = (): FolderItem => ({
+    id: `${Math.random() * 100}`,
+    name: 'PX Blue is Awesome.pdf',
+    progress: 0,
+    status: `Uploading (0%)`,
+});
+
+export const ProgressBar = (): JSX.Element => {
+    const dispatch = useDispatch();
+    const classes = useStyles();
+    const [list, setUploadFileList] = useState<FolderItem[]>(uploadFileList);
+
+    const uploadFile = useCallback((): void => {
+        setUploadFileList((oldList) => [...oldList, createFileItem()]);
```

Comment:
I don't think I mentioned why we do this when I sent you the snippet. This was something I only read about a few weeks ago. But when you are updating a state variable based on the current value of the variable, it's best practice to do it via the function as you have it here, instead of:
```
setUploadFileList([...list, createFileItem()])
```

which could pick up pending changes to the state variable from elsewhere. Using the function guarantees that you get the most up to date value for the previous value.

Nothing to change here, just thought I would give some context around that.